### DISCUSSION TITLE
reformat_media purpose, OFRenamer

### DISCUSSION THREAD

**Discussion by A-Random-Mf** [DIGITALCRIMINALS/OnlyFans#414] [Q&A]
What is the reformat_media option's function?
Is it possible to run OFRenamer separately? I want to rename my files to be %Y-%m-%d-post_id_media_id.ext instead of the current %m-%d-%Y-filename.ext

**Comment by DIGITALCRIMINAL** [comment]
Not yet, it's harder to do than I imagined. I gotta do a lot of changes to make it run separate.

**Comment by A-Random-Mf** [comment]
Ok, rather than it running independently, can you add a config option for it to run every time on all databases the script is run? I have a lot of dbs that need the directory and filename tables repopulated. (Shot myself in the foot with a script) That OFRenamer module fixes those tables for subscribed models perfectly but not expired ones

**Reply by clouds321** [reply]
I would strongly suggest [DIGITALCRIMINALS](https://github.com/DIGITALCRIMINALS) to change the default format to %Y-%m-%d

I rename a lots of file names to YYYY-MM-DD format with ZMV

ZMV only works with Z Shell, but you could use other tools I guess. 

This is the code I use

 `zmv -n '(*)([0-9][0-9]-)([0-9][0-9]-)([0-9][0-9][0-9][0-9]-)(*)' '${1}${4}${3}${2}${5}' `

This is just a "dry run" to test the results, remove -n to actually rename files. 

If you want to rename everything in subfolders 

`zmv -n '(**/)(*)([0-9][0-9]-)([0-9][0-9]-)([0-9][0-9][0-9][0-9]-)(*)' '${1}${2}${5}${4}${3}${6}' `

The pattern matching is not regex, but globbing, it looks kinda silly but works, if files are already formatted Y-M-D they will be skipped, but test for yourself.
